src/main.cpp

```diff
@@ -0,0 +1,30 @@
+#include<iostream>
+#include "vector.hpp"
+
+
+
+struct S{
+    S(int val1, int val2)
+    {
+        m_val = val1;
+        n_val = val2;
+    }
+    ~S()
+    {
+
+    }
+    int m_val;
+    int n_val;
+};
+
```

Comment:
Try this test below. It will fail at runtime.

[ As of now it will fail at compile time itself because you didn't test out all the vector member functions. You must ensure 100% code coverage when you deal with templates. Not all of your code is checked for compilation unless specific templates are used ( instantiated ).

When you fix the compilation, this test will fail at runtime.

struct S2 {
    S2() = default;
    S2& operator=(const S2&) {
        assert(x == 4);
        return *this;
    }
    int x = 4;
};

void test2() {
    Vector<S2> vec;
    vec.push_back(S2{});
    Vector<S2> vec2 = vec;
}